synedrion/src/cggmp21/sigma/dec.rs

```diff
@@ -181,29 +282,49 @@ mod tests {
 
         let aux: &[u8] = b"abcde";
 
-        // We need something within the range -N/2..N/2 so that it doesn't wrap around.
-        let y = SecretSigned::random_in_exp_range(&mut OsRng, Paillier::PRIME_BITS * 2 - 2);
-        let x = *secret_scalar_from_signed::<Params>(&y).expose_secret();
-
+        let x = SecretSigned::random_in_exponent_range(&mut OsRng, Params::L_BOUND);
+        let y = SecretSigned::random_in_exponent_range(&mut OsRng, Params::LP_BOUND);
         let rho = Randomizer::random(&mut OsRng, pk);
-        let cap_c = Ciphertext::new_with_randomizer_signed(pk, &y, &rho);
+
+        // We need $enc_0(y, \rho) = K (*) x + D$,
+        // so we can choose the plaintext of `K` at random, and derive the plaintext of `D`
+        // (not deriving `y` since we want it to be in a specific range).
+
+        let k = SecretSigned::random_in_exponent_range(&mut OsRng, Paillier::PRIME_BITS * 2 - 1);
+        let cap_k = Ciphertext::new(&mut OsRng, pk, &k);
+        let cap_d = Ciphertext::new_with_randomizer(pk, &y, &rho) + &cap_k * &-&x;
```

Comment:
`&-&x` is not Rust's proudest moment. That's a terrible syntax.

Reply:
There is a `Neg` impl for this type that consumes the value too, but we need `x` later, so the alternative would be to clone it, which wouldn't look great either.

Reply:
Yeah I tried a few things here, but this is the least worst.